elegant/process_data.py

```diff
@@ -80,6 +85,75 @@ def annotate_stage_pos(experiment_root, position, metadata, annotations):
     annotations['stage_y'] = y
     annotations['starting_stage_z'] = z
 
+def annotate_lawn(experiment_root, position, metadata, annotations, num_images_for_lawn=3):
+    '''Position annotator used to find the lawn and associated metadata about it'''
+
+    print(f'Working on position {position}')
+    position_root = experiment_root / position
+    lawn_mask_root = experiment_root / 'derived_data' / 'lawn_masks'
+    lawn_mask_root.mkdir(parents=True, exist_ok=True)
+
+    microns_per_pixel = process_images.microns_per_pixel(metadata['objective'],metadata['optocoupler'])
+
+    position_images = load_data.scan_experiment_dir(experiment_root)[position]
+    first_imagepaths = []
+    for timepoint, timepoint_images in position_images.items():
+        if position_root / (timepoint + ' bf.png') in timepoint_images:
+            first_imagepaths.append(str(position_root / (timepoint + ' bf.png')))
+        if len(first_imagepaths) == num_images_for_lawn:
+            break
+
+    first_images = list(map(freeimage.read, first_imagepaths))
```

Comment:
Generally I prefer list comprehensions to `list(map(...))` just because it's a bit more readable. Especially as you're doing comprehensions below anyway.